### ISSUE TITLE
hnsw_stats.ndis is not updated

### ISSUE THREAD

**Issue by SilvioM97** [facebookresearch/faiss#3051]
### Discussed in https://github.com/facebookresearch/faiss/discussions/3050

<div type='discussions-op-text'>

<sup>Originally posted by **SilvioM97** September 11, 2023</sup>
Hi there, is there a way to know how many elements are visited in a graph index (e.g. HNSW or NSG) during a kNN search for a query? I didn't find it so I'm asking your help. </div>

Hi there, I had this question and now I've an answer. However, I noticed that while ivf_stats.ndis is correctly updated, hnsw_stats_ndis is never updated and it is always 0. 

This is a working minimal code which reproduces the issue:

```
import faiss
import numpy as np

hnsw = faiss.IndexHNSWFlat(50, 20, faiss.METRIC_L2)
x = np.random.random((10000, 50)).astype('float32')
hnsw.add(x)
hnsw_stats = faiss.cvar.hnsw_stats
hnsw_stats.reset()
y = np.random.random((100, 50)).astype('float32')
a, b = hnsw.search(y, 5)
print(hnsw_stats.ndis)
```

**Comment by mdouze** [comment]
We may have dropped updating ndis for HNSW at some point. Let me confirm.

**Comment by mdouze** [comment]
We should re-enable populating ndis stats.

**Comment by mdouze** [comment]
I think I removed it because it is the DistanceComputer's responsibility to keep track of this, but it never did. So either the HNSW engine or the distancecomputer should do it.

**Comment by junjieqi** [comment]
@mdouze is it fixed? Should we close this one?

**Comment by mdouze** [comment]
Fixed in #3309

**Comment by matchyc** [comment]
@mdouze I think it return back to always zero... 
run pytest:
```
test_graph_based.py:134: AssertionError
=============================== short test summary info ================================
FAILED test_graph_based.py::TestHNSW::test_ndis_stats - AssertionError: 0 not greater than 8000
```
